src/cdef.rs

```diff
@@ -71,28 +76,31 @@ impl cdef::Fn {
     /// so in order to get access to pre-filter top pixels, use `top`.
     pub fn call<BD: BitDepth>(
```

Comment:
So LLVM is bad at optimizing `WithOffset` even for a normal Rust function, not just the `extern "C"` one with extra args at the end, including the weird `FFISafe` pointers?

Reply:
In my experiments LLVM doesn't always optimize them (or other small Copy structs) away even if there are only normal Rust functions involved, but never when they cross FFI boundaries (for example, if you set bottom = ptr:null() here, the extra stack usage will be optimized away, but if you pass in references to WithOffsets, used them, then call FFI with the new deconstructed parameters, they still won't be optimized away.

*I tried dozens of things so it's possible I missed something or confused some of the results so please take this with a grain of salt 🙏

Reply:
😭 That's very unfortunate.  `WithOffset` is such a simple, thin wrapper over two args.  I could see `*const FFISafe<WithOffset<T>>` not being optimized well to `*const FFISafe<T>` and `usize`, but the much simpler `WithOffset<T>` not being optimized the same as `T` and `usize` is really atrocious.